### PR TITLE
Add support for linting git commit message files

### PR DESCRIPTION
**Pull request by nickdiego**
- This makes possible to lint commit messages
- Using `gitlint` project: https://github.com/jorisroovers/gitlint

### CODE REVIEW

**Comment by nickdiego** [comment]
Hello,
As I was not familiar to the ALE codebase neither to Vim script, I implemented gitlint integration based on flake8 linter, so maybe there is still some unnecessary code in this patches.
Let me know if there's something I should remove or rearrange..
Thanks

**Comment by nickdiego** [comment]
> I think you can probably remove the parts of the code relating to running the linter with -m. That's mostly useful for flake8 for running it with different Python versions, which accept different syntax.
> 
> You can remove the version checking parts. Replace command_chain with command_callback and just the one callback instead.
> 
> You can remove the part which parses Python exceptions probably. That's needed for flake8 as it sometimes throws exceptions for syntax errors and such.
> 
> You can remove the old _args fallback, as this is a new linter.

Ok, done. :)

Thanks

File: ale_linters/gitcommit/gitlint.vim [hunk 1]
```diff
@@ -0,0 +1,50 @@
+" Author: Nick Yamane <nick.diego@gmail.com>
+" Description: gitlint for git commit message files
+
+let g:ale_gitcommit_gitlint_executable =
+\   get(g:, 'ale_gitcommit_gitlint_executable', 'gitlint')
+let g:ale_gitcommit_gitlint_options = get(g:, 'ale_gitcommit_gitlint_options', '')
+let g:ale_gitcommit_gitlint_use_global = get(g:, 'ale_gitcommit_gitlint_use_global', 0)
```
Comment (w0rp) [inline ale_linters/gitcommit/gitlint.vim]: This flag isn't used. You should either remove it from here and the documentation, or use it to find the path to the executable in virtualenv with the python function. You might want to do the latter.

Comment (nickdiego) [reply ale_linters/gitcommit/gitlint.vim]: Oops, done.

File: doc/ale.txt [hunk 2]
```diff
@@ -68,6 +68,8 @@ CONTENTS                                                         *ale-contents*
       gcc.................................|ale-fortran-gcc|
     fusionscript..........................|ale-fuse-options|
       fusion-lint.........................|ale-fuse-fusionlint|
+    git commit.............................|ale-gitcommit-options|
```
Comment (w0rp) [inline doc/ale.txt]: The helptag is misaligned here.

File: doc/ale-gitcommit.txt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  This variable can be changed to add command-line arguments to the gitlint
+  invocation.
+
+  For example, to dinamically set the gitlint configuration file path, you
+  may want to set >
+
+  let g:ale_gitcommit_gitlint_options = '-C /home/user/.config/gitlint.ini'
+<
+g:ale_gitcommit_gitlint_use_global         *g:ale_gitcommit_gitlint_use_global*
```
Comment (w0rp) [inline doc/ale-gitcommit.txt]: Put two blank lines after option sections. The line with `<` can be considered blank, as the character is concealed.

File: doc/ale.txt (hunk 2 above)
Comment (nickdiego) [reply doc/ale.txt]: Done.

File: doc/ale-gitcommit.txt (hunk 3 above)
Comment (nickdiego) [reply doc/ale-gitcommit.txt]: Ok. Done.

**Pull request merged**